homeassistant/components/mqtt/mixins.py

```diff
@@ -311,12 +311,19 @@ async def _async_setup_entities() -> None:
             # The platform has been reloaded
             config_yaml = mqtt_data.updated_config
```

Comment:
Not related to this PR, but I think there's a bug here?

If the MQTT config is emptied by the user, we'll setup the entities which were in configuration.yaml when core was started because because `mqtt_data.updated_config` is falsy?

Reply:
It is a left over from the complex config management we had before. It only is used when reloading yaml. I'll open a PR to clean this up.

Reply:
#96812